src/Modules/OrchardCore.Commerce/Services/PriceVariantProvider.cs

```diff
@@ -94,4 +94,14 @@ private ShoppingCartItem AddPriceToShoppingCartItem(ShoppingCartItem item, Produ
 
         return null;
     }
+
+    public async Task<bool> IsApplicableAsync(IList<ShoppingCartItem> items)
+    {
+        var skuProducts = await _productService.GetSkuProductsAsync(items);
+
+        return items
+            .All(item => (skuProducts.TryGetValue(item.ProductSku, out var productPart) &&
+                productPart.ContentItem.OfType<PricePart>().Any()) ||
+                productPart.ContentItem.OfType<PriceVariantsPart>().Any());
```

Comment:
I don't understand this. Why are you checking for `PricePart` and why is it applicable if you have it? I think `PriceVariantProvider` is only relevant if you have `PriceVariantsPart`. A content type shouldn't have both parts anyway.

Also you put the parenthesis on the wrong place so we have this:
![image](https://user-images.githubusercontent.com/4007293/183973383-e18dd187-1c95-4442-9f84-ff3b0145e4a9.png)

Reply:
`PriceVariantProvider` in my experience can handle content items with `PricePart` too that's why I added both, so it's technically "applicable", but you are right.